sonar-db/src/main/java/org/sonar/db/component/ComponentDao.java

```diff
@@ -297,12 +288,16 @@ public void selectForIndexing(DbSession session, @Nullable String projectUuid, R
   private static void addPartialQueryParameterIfNotNull(Map<String, Object> parameters, @Nullable String keyOrNameFilter) {
     // TODO rely on resource_index table and match exactly the key
     if (keyOrNameFilter != null) {
-      parameters.put("query", "%" + keyOrNameFilter.toUpperCase(Locale.ENGLISH) + "%");
+      parameters.put("query", queryParameterFrom(keyOrNameFilter));
     }
   }
 
-  private static void addProjectQualifier(Map<String, Object> parameters) {
-    parameters.put("qualifier", Qualifiers.PROJECT);
+  @CheckForNull
+  private static String queryParameterFrom(@Nullable String keyOrNameFilter) {
+    if (keyOrNameFilter != null) {
+      return "%" + keyOrNameFilter.toUpperCase(Locale.ENGLISH) + "%";
```

Comment:
should be replaced by `DatabaseUtils.buildLikeValue(keyOrNameFilter.toUpperCase(LocalEnglish))` so that value is correctly escaped. Note that the SQL clause must also be updated by adding `escape '/'` (see javadoc of `buildLikeValue()`